### ISSUE TITLE
different target score when using logits from sample_autoregressive

### ISSUE THREAD

**Issue by marton-avrios** [tensorflow/mesh#262]
I am currently try to implement returning logits along with prediction from `sample_autoregressive` to calculate score from them. However the score calculated from these logits are slightly different from the ones I get with `mode="scoring"`. Any explanation?
